coderd/healthcheck/healthcheck.go

```diff
@@ -153,6 +174,9 @@ func Run(ctx context.Context, opts *ReportOptions) *Report {
 	if !report.Database.Healthy {
 		report.FailingSections = append(report.FailingSections, SectionDatabase)
 	}
+	if !report.WorkspaceProxy.Healthy {
+		report.FailingSections = append(report.FailingSections, SectionWorkspaceProxy)
```

Comment:
side thought: frontend does not use `FailingSections` at all. Maybe we should remove it for simplicity.

Reply:
👍 Will file a follow-up PR

Reply:
I think it's beneficial to have a structure that can be navigated in a generic way.

```js
for (let section of report.failing_sections) {
	console.log(report[section].reason);
}
```

Given a good design, it will "always works", even when new fields and structures are added. Granted, I'm not a proponent for the current design, just for the ability to do so.